columnflow/plotting/plot_util.py

```diff
@@ -314,6 +314,15 @@ def apply_variable_settings(
     return hists, variable_style_config
 
 
+def remove_negative_contributions(hists: dict[Hashable, hist.Hist]) -> dict[Hashable, hist.Hist]:
+    for proc_inst, h in list(hists.items()):
+        h = h.copy()
+        h.view().value[h.view().value < 0] = 0
+        h.view().variance[h.view().variance < 0] = 0
+        hists[proc_inst] = h
+    return hists
```

Comment:
To be consistent with the other methods, we shouldn't adjust the input arguments in-place. The copy() below is shallow.


```suggestion
    _hists = hists.copy()
    for proc_inst, h in hists.items():
        h = h.copy()
        h.view().value[h.view().value < 0] = 0
        h.view().variance[h.view().variance < 0] = 0
        _hists[proc_inst] = h
    return _hists
```